Release note for 2024-01-08 (#4051)
Azure/AKS: CHANGELOG.md

* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.

> * Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.

Reviewer: should this go under behavior change?

Reviewer: "* Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads."

Author: This one was inherited from the previously announcement. Though I do agree it fits behavior change more, updated

* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
* Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024. See [release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar:~:text=patched%20and%20supported.-,AKS%20Kubernetes%20release%20calendar,-View%20the%20upcoming).
 
### Release notes

* Bug Fixes 

>   * Fix for [LTS](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#long-term-support-lts) to look at AgentPoolResources.

Reviewer 2: - need to remove 'en-us'
- needs more context for what/where customer is looking for with respect to agent pool and LTS

Author: Confirmed customer won't notice such change, removed

  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.

>   * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.

Reviewer 2: ```suggestion
  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used resulting in potential failures in enabling the KEDA addon. This issue has now been fixed.
```

* Behavioral Change

>   * [Artifact Streaming](https://learn.microsoft.com/en-us/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.

Reviewer 2: ```suggestion
  * [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing node pool.
```

  * The default ResponseHeaderTimeout (The timeout of http client used by AKS RP, which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased from 200 Mi to 300Mi.

>   * [Expanders flag is removed](https://github.com/Azure/azure-rest-api-specs/pull/26440?expand=1&template=data_plane_template.md) from [AutoscalerProfile](https://learn.microsoft.com/en-us/azure/aks/hybrid/work-with-autoscaler-profiles#profile-settings) from 2023-11-01-preview since it may cause confusion with existing Expanders. 

Reviewer 2: en-us needs to be removed

Author: Updated

# Azure Kubernetes Service Changelog

> ## Release 2024-01-08

Reviewer 3: Do we want to update this to whatever date this is published?

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements

* Kubernetes 1.25 is being deprecated on January 14, 2024 and support will transition to our [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy). Please upgrade to Kubernetes version 1.26 or above.

> * Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks:~:text=exception%20YAML.-,Install%20Azure%20Policy%20Add%2Don%20for%20AKS,-Before%20you%20install) will now no longer support this. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/).

Reviewer 3: > will now no longer support this.

Do we want to be more explicit and spell out "this" so people don't need to try to figure out what exactly applies?

Author: Updated to replace with 'validation for constraint template'

> * Starting Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.

Author: Updated

> * Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.

Author: Updated, MB

> * Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.

Reviewer 4: ```suggestion
* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024. See [release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar:~:text=patched%20and%20supported.-,AKS%20Kubernetes%20release%20calendar,-View%20the%20upcoming).
```

Author: Updated

 
### Release notes

* Bug Fixes 

>   * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.

Reviewer 4: I'm checking with eng on this one to see if we want to reword it

Reviewer 4: Update: Remove this
```suggestion
```

Author: Updated, thanks for verifying

>   * Fix for LTS to look at AgentPoolResources.

Reviewer 4: Link to LTS docs? And how does that impact customers?

Author: Updated

>   * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.

Reviewer 2: ```suggestion
  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
```

Author: Updated, thanks for rewording

>   * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.

Author: Updated, we didn't see any keda usage at any cloud environment except public cloud before. Thus, theoretically the wrong image URL will cause fail enablement.

> * Behavioral Change

Reviewer 4: [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.

Reviewer 5: Wherever possible please add GH work items translating to the bug .

Author: Added

>   * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information and is generated from ManagedCluster abnd AgentPool. Thus, they should no longer be used.

Reviewer 5: Any doc link talking about this?

Reviewer 2: from a customer's perspective, what are we referring to with respect to 'goals'? something related to API?

Author: Confirmed with Hao Yuan, this change should be considered as internal change and no customer awared impact would be caused, Removed

>   * The default ResponseHeaderTimeout (which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.

Reviewer 2: Sharing questions on this in internal thread

Author: Kindly see the reply in the thread

>   * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.

Reviewer 2: increased from?

>   * Expanders flag is removed from AutoscalerProfile from November API

Reviewer 5: Perhaps some doc reference to autoscalerprofile would help.

Reviewer 2: - what impact or behavior change does this removal have? does this impact customers with existing clusters in any way?
- for API version, we need to mention the official format. for example, 2023-11-01, 2023-11-01-preview,...

Author: It is basically a going back to the old behavior and removing the field we added. It's breaking but it's in preview only so it's OK according to Matthew.

>   * The default ResponseHeaderTimeout (The timeout of http client used by AKS RP, which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.

Reviewer 6: What's the "before" behavior for reference of the audience?

Author: There wasn't a ResponseHeaderTimeout set before, thus this is a new behavior

>   * Expanders flag is removed from [AutoscalerProfile](https://learn.microsoft.com/en-us/azure/aks/hybrid/work-with-autoscaler-profiles#profile-settings) from 2023-11-01-preview since it may cause confusion with existing Expanders. 

Reviewer 6: How does this impact existing users that already uses Expander?

Author: It's going to be a breaking change, it is still in preview though

Reviewer 6: @charleswoll ack. Maybe you want to add it doesn't affect current usage of "expander" API call to avoid panic like I had?

  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.
    
* Component Updates
  * Windows Kubernetes CVE fixes for [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
  * Update ama-logs addon image to [3.1.16](https://github.com/microsoft/Docker-Provider/blob/ci_prod/ReleaseNotes.md)

>   * Azure Windows 2019 Image has been updated to [Azure Windows - 17763.5206.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2019/17763.5206.231213.txt)

Reviewer 5: Why is AZ linux (mariner) VHD missing here ?https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSCBLMarinerV2/gen2/202401.03.0.txt

Author: Thought we should be watching for AzureLinux, added and updated

>   * Azure Windows 2019 Image has been updated to [Azure Windows - 17763.5206.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2019/17763.5206.231213.txt)

Author: updated

>   * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.

Reviewer 6: Which feature it is for?

Author: No sensible customer impact, thus removed

>   * Azure Windows 2022 Image has been updated to [Azure Windows - 20348.2159.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2022/20348.2159.231213.txt)

Reviewer 4: ```suggestion
  * Windows Server 2022 Image has been updated to [20348.2159.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2022/20348.2159.231213.txt)
```

Author: Updated

 
### Release notes

* Preview Features
   *  Starting January 2024, the [AI toolchain operator](https://learn.microsoft.com/azure/aks/ai-toolchain-operator) for AKS can now be enabled to set up the GPU infrastructure and deploy OSS models from the Azure CLI in a matter of minutes.
   *  [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing node pool.
  
* Bug Fixes 
  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used resulting in potential failures in enabling the KEDA addon. This issue has now been fixed.

* Behavioral Change

>   * Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.

Reviewer 7: Consider linking to this issue I just filed which also talks about this: #4060

* Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.
 
### Release notes

* Bug Fixes 
  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
  * Fix for LTS to look at AgentPoolResources.
  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.

* Behavioral Change

>   * ManagedCluster, AgentPool are now deprecated fields in agentpool goals.

Reviewer 2: goals?

Author: Updated

  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information. It is genrated from ManagedCluster abnd AgentPool and they should no longer be used.

Reviewer 2: From a customer's perspective, what goals are we referring to? something related to API?

Author: Confirmed with Hao Yuan, this one should be removed as it is a internal only change

>   * The default ResponseHeaderTimeout is now set to 55s to prevent abnormal slow operation caused by hanging connection.

Reviewer 2: Needs more context. Response header timeout for?

>   * The default ResponseHeaderTimeout is now set to 55s to prevent abnormal slow operation caused by hanging connection.

Reviewer 2: Sharing questions on this in internal thread

>   * Community images are now added into reduction pipelines for cost saving.

Reviewer 2: Reduction pipelines? Context? Is this external facing?

>   * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.

Reviewer 2: - need to remove 'en-us'. 
- Increased from what value to 300Mi?

Author: updated

>   * Expanders flag is removed from AutoscalerProfile.

Reviewer 2: Context? From which API version? Any next steps for users having this on their existing clusters?

Author: Updated

>   * Config checksum is added to cns daemonset spec.

Reviewer 2: Context? Why was this change introduced? To rollover daemonset when ConfigMap is changed?

Author: Updated
  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.

>   * Orchestrator version is moved to new validator removing from server/validation.

Reviewer 2: Is this external facing?

Author: Removed

* Component Updates

>   * Windows Kubernetes CVE fixes for 3P Windows customers, for: [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)

Reviewer 2: ```suggestion
  * Windows Kubernetes CVE fixes for [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
```

>   * Update ama-logs addon image to 3.1.16

Reviewer 2: any link to release notes?

Author: updated

  * Bumpe cloud-controller-manager image to [v1.25.23](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.25.23), [v1.26.18](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.18), [v1.27.12](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.12), [v1.28.4](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.4).

>   * Bump agentbaker version to v0.20231122.3, v0.20231206.0.

Reviewer 2: We can remove this line and we just need to cover the vhd notes

Author: Updated
